src/lex/mod.rs

```diff
@@ -279,10 +279,7 @@ impl Lexer {
                 if radix == 8 || radix == 10 || self.peek() == Some(b'.') {
                     start
                 } else {
-                    return Err(format!(
-                        "missing digits to {} integer constant",
-                        if radix == 2 { "binary" } else { "hexadecimal" }
-                    ));
+                    return Err(LexError::MissingDigits(radix.try_into().unwrap()));
```

Comment:
Could you make `radix` a `Radix` instead of storing it as a u8? Then we wouldn't have this `unwrap()`. You might have to make a `.as_u8()` function for when the lexer multiplies by the radix.

Reply:
Should I move it out of the `error` module then?

Reply:
Yes, please do.